node/src/blockchain/blockchain_bridge.rs

```diff
@@ -1287,26 +1553,18 @@ mod tests {
                 },
```

Comment:
I'm commenting on the test "handle_retrieve_transactions_uses_latest_block_number_upon_get_block_number_error".
Can you explain me of what use is this cannibalized test? 

Do you know what you did?

You changed the prod code so that there is a method `retrieve_transactions` in the BlockchainInterface which hosts some code that also includes the incriminated call of `get_block_number` which is linked to the lower level interface. You modified it slightly but the test still is using the BlockchainInterfaceMock. Now, isn't there anything suspicious to you having the two piece of information I just put forth to you? 

The issue is that if you use the mock version, everything out of the guts, will never be called, and you have the method, which is this test supposed to be aimed at, missing this way. So you're testing a situation which you can only get closer to in the test, but you stay away from it. 

This what I see makes me think that you only fixed a failing test but didn't think beyond it, about what the test is testing. That's a bit sad finding. 

What am I advising? Hmm. It depends on whether you've had already the discussion with Dan about the right and durable design of the BlockchainInterface, because there might be some rearranging. 

Otherwise, if you cannot test the condition from this level, but the condition is still there, you have two options.
First, I just recalled **why I designed the LowerLevelInt in first place**. It was so that I could take the real implementation of the higher level (or main) interface, planning that it will engage in the test but **yet one component inside would need to be altered** to a mockable object. That's the reason, why the lower level interface has its own trait. So that I could mock it out if I fancied. This was to solve the big problem of testing these complex, layered structures. 

Another way is to write this with the BlockchainInterfaceReal and use your MockBlockchainClientServer to invoke an error on the call for `get_block_number()`. You've done something like this many times already, so you might want to chose the second path by intuition.    

Nevertheless, I wonder if the old good purpose of the LowLevelInt and the possibilities of use it offered will be missing. So far, you've succeeded in a substitution of the concept by your tedious way of setting up the MBCServer. Maybe that'll be universal enough also into the future. I hope so.

Reply:
This is interconnected with a different comment now. So it shouldn't end up in the dust, unless you ignore also the other, parental comment.